new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItVoyagerSample.java

```diff
@@ -0,0 +1,298 @@
+  private static String opNamespace = null;
+  private static String domainNamespace = null;
+  private static ConditionFactory withStandardRetryPolicy = null;
+
+  private static HelmParams voyagerHelmParams = null;
+
+  /**
+   * Install operator, create a one cluster domain and install Voyager.
+   *
+   * @param namespaces list of namespaces created by the IntegrationTestWatcher by the
+   *                   JUnit engine parameter resolution mechanism
+   */
+  @BeforeAll
+  public static void init(@Namespaces(3) List<String> namespaces) {
+    // create standard, reusable retry/backoff policy
+    withStandardRetryPolicy = with().pollDelay(2, SECONDS)
+      .and().with().pollInterval(10, SECONDS)
+      .atMost(5, MINUTES).await();
+
+    // get a unique Voyager namespace
+    logger.info("Get a unique namespace for Voyager");
+    assertNotNull(namespaces.get(0), "Namespace list is null");
+    String voyagerNamespace = namespaces.get(0);
+
+    // get a unique operator namespace
+    logger.info("Get a unique namespace for operator");
+    assertNotNull(namespaces.get(1), "Namespace list is null");
+    opNamespace = namespaces.get(1);
+
+    // get a unique domain namespace
+    logger.info("Get a unique namespace for WebLogic domain");
+    assertNotNull(namespaces.get(2), "Namespace list is null");
+    domainNamespace = namespaces.get(2);
+
+    final String cloudProvider = "baremetal";
+    final boolean enableValidatingWebhook = false;
+
+    // install and verify Voyager
+    voyagerHelmParams =
+      installAndVerifyVoyager(voyagerNamespace, cloudProvider, enableValidatingWebhook);
+
+    // install and verify operator
+    installAndVerifyOperator(opNamespace, domainNamespace);
+
+    // create and verify one cluster domain
+    logger.info("Create domain and verify that it's running");
+    createAndVerifyDomain();
+  }
+
+  @AfterAll
+  void tearDown() {
+    // uninstall Voyager
+    if (voyagerHelmParams != null) {
+      assertThat(uninstallVoyager(voyagerHelmParams))
+          .as("Test uninstallVoyager returns true")
+          .withFailMessage("uninstallVoyager() did not return true")
+          .isTrue();
+    }
+  }
+
+  /**
+   * The test invokes a webapp to verify the Voyager is installed successfully and ready to use.
+   */
+  @Test
+  @DisplayName("Create the Voyager ingress and invoke the webapp to verify Voyager load balancer works")
+  @Slow
+  @MustNotRunInParallel
+  public void testVoyagerWorks() {
+    String ingressName = domainUid + "-ingress-host-routing";
+    String ingressServiceName = VOYAGER_CHART_NAME + "-" + ingressName;
+    String channelName = "tcp-80";
+
+    try {
+      // create Voyager ingress resource
+      Map<String, Integer> clusterNameMsPortMap = new HashMap<>();
+      clusterNameMsPortMap.put(clusterName, managedServerPort);
+      List<String>  hostNames =
+          installVoyagerIngressAndVerify(domainUid, domainNamespace, ingressName, clusterNameMsPortMap);
```

Comment:
check Xian's PR https://github.com/oracle/weblogic-kubernetes-operator/pull/1740/ which checks for ingress ready

Reply:
isn't the ingress creation same for Nginx or Voyager?

Reply:
No They are diff

Reply:
We don't know the Nodeport until the Voyager ingress controller is created

Reply:
I will try to create a generic method that is not related to either Nginx or Voyager, see if we can consolidate

Reply:
Voyager ingress controller is already created, you can pass the node port to this method and verify ingress is ready by accessing ready app?

Reply:
No, I need to get Nodeport after ingress controller is cfeated in installVoyagerIngressAndVerify, not passed to the method

Reply:
ok, I think we are using different terms here but talking the same ...ingress controllers are nginx, voyager, traefik, etc ..ingress resource is what you are creating in this method. You mean you can access node port after creating the ingress resource?

Reply:
all clear after discussing in slack, huiling will make the needed changes. Thanks Huiling.

Reply:
fixed